junit-jupiter-api/src/main/java/org/junit/jupiter/api/ClassOrderer.java

```diff
@@ -0,0 +1,230 @@
+	 * effectively cause them to appear at the end of the sorted list, unless
+	 * certain classes are assigned an explicit order value greater than the default
+	 * order value. Any classes assigned an explicit order value greater than the
+	 * default order value will appear after non-annotated classes in the sorted
+	 * list.
+	 */
+	class OrderAnnotation implements ClassOrderer {
+
+		/**
+		 * Sort the classes encapsulated in the supplied
+		 * {@link ClassOrdererContext} based on the {@link Order @Order}
+		 * annotation.
+		 */
+		@Override
+		public void orderClasses(ClassOrdererContext context) {
+			context.getClassDescriptors().sort(comparingInt(OrderAnnotation::getOrder));
+		}
+
+		private static int getOrder(ClassDescriptor descriptor) {
+			return descriptor.findAnnotation(Order.class).map(Order::value).orElse(Order.DEFAULT);
+		}
+	}
+
+	/**
+	 * {@code ClassOrderer} that orders classes pseudo-randomly.
+	 *
+	 * <h4>Custom Seed</h4>
+	 *
+	 * <p>By default, the random <em>seed</em> used for ordering classes is the
+	 * value returned by {@link System#nanoTime()} during static initialization
+	 * of this class. In order to support repeatable builds, the value of the
+	 * default random seed is logged at {@code INFO} level. In addition, a
+	 * custom seed (potentially the default seed from the previous test plan
+	 * execution) may be specified via the {@link Random#RANDOM_SEED_PROPERTY_NAME
+	 * junit.jupiter.execution.class.order.random.seed} <em>configuration parameter</em>
+	 * which can be supplied via the {@code Launcher} API, build tools (e.g.,
+	 * Gradle and Maven), a JVM system property, or the JUnit Platform configuration
+	 * file (i.e., a file named {@code junit-platform.properties} in the root of
+	 * the class path). Consult the User Guide for further information.
+	 *
+	 * @see Random#RANDOM_SEED_PROPERTY_NAME
+	 * @see java.util.Random
+	 */
+	class Random implements ClassOrderer {
+
+		private static final Logger logger = LoggerFactory.getLogger(Random.class);
+
+		/**
+		 * Default seed, which is generated during initialization of this class
+		 * via {@link System#nanoTime()} for reproducibility of tests.
+		 */
+		private static final long DEFAULT_SEED;
+
+		static {
+			DEFAULT_SEED = System.nanoTime();
+			logger.info(() -> "ClassOrderer.Random default seed: " + DEFAULT_SEED);
+		}
+
+		/**
+		 * Property name used to set the random seed used by this
+		 * {@code ClassOrderer}: {@value}
+		 *
+		 * <h3>Supported Values</h3>
+		 *
+		 * <p>Supported values include any string that can be converted to a
+		 * {@link Long} via {@link Long#valueOf(String)}.
+		 *
+		 * <p>If not specified or if the specified value cannot be converted to
+		 * a {@link Long}, the default random seed will be used (see the
+		 * {@linkplain Random class-level Javadoc} for details).
+		 */
+		public static final String RANDOM_SEED_PROPERTY_NAME = "junit.jupiter.execution.class.order.random.seed";
```

Comment:
Sorry, one more thing... this is untested and I'm wondering if it shouldn't be the same seed as for method ordering.

@junit-team/junit-lambda @ledoyen WDYT?

Reply:
I definitely think we need a single "random seed" for any such random ordering of containers and tests.